packages/ui/src/working-groups/components/Activities/ApplicationWithdrawnContent.tsx

```diff
@@ -1,24 +1,17 @@
 import React from 'react'
 import { Link } from 'react-router-dom'
 
-import { useModal } from '@/common/hooks/useModal'
+import { MemberModalLink } from '@/common/components/ModalLink'
 
 import { ApplicationWithdrawnActivity } from '../../types'
 
 interface Props {
   activity: ApplicationWithdrawnActivity
 }
 
-export const ApplicationWithdrawnContent = ({ activity }: Props) => {
-  const { showModal } = useModal()
-  const { membership, opening } = activity
-  return (
-    <>
-      <Link to="#" onClick={() => showModal({ modal: 'Member', data: { id: membership.id } })}>
-        {membership.handle}
-      </Link>{' '}
-      has withdrawn application from "<Link to={`/working-groups/openings/${opening.id}`}>{opening.title}</Link>"
-      opening.
-    </>
-  )
-}
+export const ApplicationWithdrawnContent = ({ activity: { membership, opening } }: Props) => (
+  <>
+    <MemberModalLink call={{ modal: 'Member', data: { id: membership.id } }}>{membership.handle}</MemberModalLink> has
+    withdrawn application from "<Link to={`/working-groups/openings/${opening.id}`}>{opening.title}</Link>" opening.
```

Comment:
So far I went with C, I'm also partial to B despite it being quite messy indeed. I just feel like the type/component itself should enforce the types of arguments it accepts. It may be strictness for strictness' sake though

Reply:
I think that this one might be temporary nevertheless #613

Reply:
I think that this one might be temporary nevertheless #613